packages/core/lib/modules/accounts.js

```diff
@@ -2,63 +2,126 @@ import {WalletRpc} from '../client/wallet-rpc';
 import {UtilCryptoRpc} from '../client/util-crypto-rpc';
 import {KeyringRpc} from '../client/keyring-rpc';
 import {ApiRpc} from '../client/api-rpc';
-import {Wallet} from './wallet';
-import { Errors } from '../errors';
-
-export type KeypairType = 'sr25519' | 'ed25519' |  'ecdsa';
+import {Wallet, WalletEvents} from './wallet';
+import {Errors} from '../errors';
+import {KeypairType, WalletDocument, DocumentType} from '../types';
+import {EventManager} from './event-manager';
 
 export type Account = {
-  id: string;
-  name: string;
-  type: KeypairType;
-  address: string;
-}
+  id: string,
+  name: string,
+  type: KeypairType,
+  address: string,
+  correlations: any[],
+};
+
+export const AccountsEvents = {
+  loaded: 'loaded',
+  accountCreated: 'account-added',
+  accountUpdated: 'account-updated',
+  accountRemoved: 'account-removed',
+};
 
 // TODO: Add events to accounts
 export class Accounts {
   accounts: Account[];
   wallet: Wallet;
+  eventManager: EventManager;
+  static DocumentFilters = {
+    mnemonicType: (item: WalletDocument) => item.type === 'Mnemonic',
+    currencyType: (item: WalletDocument) => item.type === 'Currency',
+  };
 
   constructor() {
     this.accounts = [];
     this.wallet = Wallet.getInstance();
+    this.eventManager = new EventManager();
+    this.eventManager.registerEvents(AccountsEvents);
   }
 
   async load() {
     this.accounts = await Wallet.getInstance().query({
-      type: 'account',
+      type: 'Address',
     });
-
+    this.eventManager.emit(AccountsEvents.loaded);
     return this.accounts;
   }
 
-  getBalance(address) {
-    return ApiRpc.getAccountBalance(accountId);
+  async exportAccount(accountId, password) {
+    return WalletRpc.exportAccount(accountId, password);
+  }
+  
+  async importAccount(json, password) {
+    return this.create({
+      json,
+      password,
+    })
+  }
+
+  async fetchBalance(accountId) {
+    const account = await this.getAccount(accountId);
+    const balance = await ApiRpc.getAccountBalance(account.address);
+    const currency = await this.findCorrelationByType(accountId, 'Currency');
+    
+    currency.value = balance;
+    
+    await this.wallet.update(currency);
+   
+    return balance;
+  }
+  
+  async getBalance(accountId) {
+    const currency = await this.findCorrelationByType(accountId, 'Currency');
+    return currency.value;
   }
 
   getAccounts() {
     return this.accounts;
   }
 
+  async getAccount(accountId): Promise<Account> {
+    const accountDocument = await Wallet.getInstance().getDocumentById(
+      accountId,
+    );
+
+    return {
+      ...accountDocument,
+      correlations: await WalletRpc.resolveCorrelations(accountId),
+    };
+  }
+
+  async findCorrelationByType(accountId: string, type: DocumentType) {
+    const account = await this.getAccount(accountId);
```

Comment:
feels like you might want to add a `validateAccountId(accountId)` function in here so you do consistent parameter validation throughout.

Reply:
fixed